src/shims/fs.rs

```diff
@@ -371,10 +371,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         this.check_no_isolation("fstat")?;
 
-        if this.tcx.sess.target.target.target_os.to_lowercase() != "macos" {
```

Comment:
Uh... what makes you so confident as to remove these checks?
We should have a helper method for them though I think.

Reply:
It might be a little bit fragile, but `fstat` is dispatched under the `macos` module only.

Reply:
Yeah that's too fragile, please keep the checks here.

Also it might be a good convention to make sure all such platform-specific methods have the platform in their name, like `macos_fstat` or so.